### PR TITLE
Overmap Extension Part 1

### PR DESCRIPTION
**Pull request by hyperioo**
## About The Pull Request

Implementation of the following Kot map items:
* Star system map
* Port Planetary Exploration from Eclipse
* Locators and scanning (partly done)

## Why It's Good For The Game

Improved overmap experience:
*  you only see scanned events near the ship and not everything like before
* neat sprites

Long range scanner:
* add interactions between command and technomancers
* interaction with Moebius could be developed but is not done yet (possibility to upgrade the scanner with RnD stuff)

Planetary exploration:
* Several random ruins to explore
* A new playground for miners or people that want to go on an adventure / build their mini-fort there.

Base for future space combat and more scanner features.

Hulk and Vasiliy Dokuchaev shuttles are no longer limited in range and can go everywhere to avoid depending on Command to explore the system and discover places (it's a problem when there is no active command).

Several star types are implemented but they all have the same effect for now (drain more energy to maintain shield the closer Eris is). The star is always at the center of the overmap.

Several sprites close to each other (it's not exactly like that in game):

It looks like that:

Long range scanner:

Reworked shield generator:

Sprites from INFRARED_BARON
[Planetary exploration](https://github.com/Eclipse-Station/NEV-Northern-Light/pull/13) by drexample from Eclipse

## Changelog
:cl: Hyperio
tweak: Increased overmap size from (50, 50) to (200, 200)
tweak: Increased number of overmap events to keep the same density on overmap
add: New overmap sprites
add: New overmap scanning system
tweak: Hulk and Vasiliy Dokuchaev have now infinite range to avoid situations in which command is not active
add: Star at the center of the overmap
add: Long range scanner machinery in Section 5 to scan the area around the ship on overmap
add: Port Planetary Exploration from Eclipse
fix: Fix planetary ruins to be compatible with Eris' objects
add: New shield generator sprites + glows when turned on
tweak: Landmark naming system modified to be compatible with new overmap
/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
pog

**Comment by TheShown911** [comment]
also conflicts

**Comment by hyperioo** [comment]
I know, gonna fix them this afternoon :)

**Comment by TheShown911** [comment]
:)

File: code/__DEFINES/culture.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+#define FACTION_HEPHAESTUS     "Hephaestus Industries"
+#define FACTION_XYNERGY        "Xynergy"
+#define FACTION_FLEET          "SCG Fleet"
+#define FACTION_EXPEDITIONARY  "Expeditionary Corps"
+#define FACTION_PCRC           "Proxima Centauri Risk Control"
+#define FACTION_CORPORATE       "Expeditionary Corps Organisation"
+#define FACTION_DAIS           "Deimos Advanced Information Systems"
+#define FACTION_OTHER          "Other Faction"
+#define FACTION_TEST_SUBJECTS  "Test Subjects"
+#define FACTION_SPACECOPS  "Sol Federal Police"
```
[the remark this answers is not in the document]
Comment (infraredbaron) [reply code/__DEFINES/culture.dm:27]: main ship designs, excluding derelicts and unknown:
image

**Comment by hyperioo** [comment]
Finally fixed this mess :agony: Just because I messed with map merger during my first commits...

**Comment by hyperioo** [comment]
Need to change update_icon and overlays += to be consistent with #5940

File: code/modules/long_range_scanner/long_range_scanner.dm [hunk 2]
```diff
@@ -0,0 +1,482 @@
+#define EVENT_ENABLED 			3
+#define EVENT_DISABLED 			4
+#define EVENT_RECONFIGURED		5
+#define PASSIVE_SCAN_RANGE      3
+#define PASSIVE_SCAN_PERIOD     3 SECONDS
```
Comment (TheShown911) [inline code/modules/long_range_scanner/long_range_scanner.dm:5]: Why don't the values have the same distance
emoji

File: code/modules/overmap/exoplanets/planet_types/snow.dm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+/datum/random_map/noise/exoplanet/snow
+	descriptor = "frozen exoplanet"
+	smoothing_iterations = 1
+	flora_prob = 5
+	large_flora_prob = 10
+	water_level_max = 3
+	land_type = /turf/simulated/floor/exoplanet/snow
+	water_type = /turf/simulated/floor/exoplanet/ice
+	//fauna_types = list(/mob/living/simple_animal/hostile/retaliate/beast/samak, /mob/living/simple_animal/hostile/retaliate/beast/diyaab, /mob/living/simple_animal/hostile/retaliate/beast/shantak)
+	//megafauna_types = list(/mob/living/simple_animal/hostile/retaliate/giant_crab)
```
Comment (TheShown911) [inline code/modules/overmap/exoplanets/planet_types/snow.dm:32]: We will need more mobs for the exoplanets due to removal of random simple animals, but that's for the future

File: code/modules/reagents/reagents/random.dm [hunk 4]
```diff
@@ -0,0 +1,158 @@
+
+
+// subtypes of stuff in here will be avoided when randomizing interactions.
```
Comment (TheShown911) [inline code/modules/reagents/reagents/random.dm:3]: Is this file included?

File: code/modules/reagents/reagents/toxins.dm [hunk 5]
```diff
@@ -653,3 +653,28 @@
...
+
+/datum/reagent/toxin/biomatter
+	name = "Biomatter"
+	id = "biomatter"
+	description = "A goo of unknown to you origin. Its better to stay that way."
+	taste_description = "vomit"
+	reagent_state = LIQUID
+	color = "#527f4f"
+	strength = 0.3
+
```
Comment (TheShown911) [inline code/modules/reagents/reagents/toxins.dm:665]: This is already there on lines 623-631

**Review by TheShown911 — APPROVED** [review]
Many missing newlines and some strange things, but overall can be merged.

File: code/modules/reagents/reagents/random.dm (hunk 4 above)
Comment (hyperioo) [reply code/modules/reagents/reagents/random.dm:3]: Hmm, indeed it's not, it must have been added when I grabbed the PR from Eclipse.

I'll remove it. random_effects.dm as well

File: code/modules/reagents/reagents/toxins.dm (hunk 5 above)
Comment (hyperioo) [reply code/modules/reagents/reagents/toxins.dm]: I've removed them

File: code/modules/long_range_scanner/long_range_scanner.dm (hunk 2 above)
Comment (hyperioo) [reply code/modules/long_range_scanner/long_range_scanner.dm:5]: There are aligned in Visual Studio Code ;-;

File: code/modules/reagents/reagents/random.dm (hunk 4 above)
Comment (hyperioo) [reply code/modules/reagents/reagents/random.dm]: Done

**Pull request merged**